packages/rearch/lib/rearch.dart

```diff
@@ -225,6 +228,28 @@ class CapsuleContainer implements Disposable {
       manager.dispose();
     }
   }
+
+  /// Warms up the specified capsules by waiting for them to all be [AsyncData].
+  Future<void> warmUp<T extends AsyncValue<dynamic>>(
```

Comment:
This function shouldn't use async/await and instead use `future.whenComplete(dispose)`

This will look like:

```dart
final dispose = ...;
return completer.future.whenComplete(dispose);
```

(and then remove the async keyword from the function declaration)